### PR TITLE
bug fixes

### PR DESCRIPTION
**Pull request by Faf4a**
## Type
- [x] Bug Fix
- [ ] Functions
- [ ] Callbacks
- [ ] Handlers
- [ ] Others

Dependencies (Third Party Modules) needed: no

Want a credit? Discord tag or other social media link: no

Referenced Issue: no

## Description
fixed getleaderboardinfo, guildcreatepost and cachemembers as always test before accepting

### CODE REVIEW

**Comment by Faf4a** [comment]
(cachemembers returnCount didnt work)
(guildcreatepost didnt work without parser)
(getleaderboardinfo returned invalid data)

**Comment by Faf4a** [comment]
@Leref hurry up !

File: src/functions/variables/getLeaderboardInfo.js [hunk 1]
```diff
@@ -14,39 +23,33 @@ module.exports = async (d) => {
...
   switch (format) {
     case "top":
-      data.result = foundDataMap.find((x) => x.key === key)?.pos || null;
+      data.result = all.slice().sort((a, b) => b.value - a.value).findIndex((x) => x.key === key) + 1 || 0;
       break;
     case "value":
-      data.result = value;
+      data.result = all.find(x => x.key === key).value || 0;
       break;
-    case "tag":
```
Comment (Leref) [inline src/functions/variables/getLeaderboardInfo.js:48]: Other leaderboard function support tag.
All should be same

**Review by Leref — CHANGES_REQUESTED** [review]

**Comment by Leref** [comment]
I'll update all leaderboard and variable functions to remove customDB since that slows down the logic entirely.

File: src/functions/variables/getLeaderboardInfo.js (hunk 1 above)
Comment (Faf4a) [reply src/functions/variables/getLeaderboardInfo.js:48]: tag is going to be removed by discord though, returning null or 0000 wouldn make sense

Comment (Faf4a) [reply src/functions/variables/getLeaderboardInfo.js:48]: I can add tag but meh

**Comment by Faf4a** [comment]
added tag option for now

**Comment by Leref** [comment]
Update this PR to latest commits

**Pull request closed without merging**